site/src/main/java/uk/nhs/digital/ps/beans/Publication.java

```diff
@@ -217,23 +217,23 @@ public boolean isPubliclyAccessible() {
     }
 
     /*
-     * Migration: 20171208
-     * New field: publicationsystem:Attachments-v2
-     * Old field: publicationsystem:attachments
+     * Migration: 20171213
+     * New field: publicationsystem:Attachments-v3
+     * Old field: publicationsystem:Attachments-v2
      *
-     * Temporary make this getter forward and backward compatible, meaning we can deploy this code without running
-     * content migration.
+     * Temporary supporting old and new versions until all the live content is migrated to the new way
      */
+    @HippoEssentialsGenerated(internalName = PropertyKeys.ATTACHMENTS_V3)
+    public List<Attachment> getAttachments() {
+        assertPropertyPermitted(PropertyKeys.ATTACHMENTS_V3);
+
+        return getChildBeansIfPermitted(PropertyKeys.ATTACHMENTS_V3, Attachment.class);
+    }
     @HippoEssentialsGenerated(internalName = PropertyKeys.ATTACHMENTS_V2)
-    public List<HippoResourceBean> getAttachments() {
+    public List<HippoResourceBean> getAttachmentsOld() {
```

Comment:
Is there some specific reason why we need to keep the `getAttachmentsOld` as a separate getter?

Could we instead keep just the `getAttachments` and have it call `getChildBeansIfPermitted` for both the new and old property, merging results before returning?

Reply:
Created RPS-238